core/commands/swarm.go

```diff
@@ -812,3 +814,72 @@ func filtersRemove(r repo.Repo, cfg *config.Config, toRemoveFilters []string) ([
 
 	return removed, nil
 }
+
+var swarmConnMgrCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Interact with the Connection Manager.",
+		ShortDescription: `
+Display information about the current state of the connection manager.
+`,
+	},
+	Run: func(req cmds.Request, res cmds.Response) {
+		n, err := req.InvocContext().GetNode()
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		if n.PeerHost == nil {
+			res.SetError(errNotOnline, cmds.ErrClient)
+			return
+		}
+
+		infos := make(map[string]interface{})
+		cmgr := n.PeerHost.ConnManager()
+		switch cmgr := cmgr.(type) {
+		case nil:
+			infos["type"] = "<nil>"
+		case *ifconnmgr.NullConnMgr:
+			infos["type"] = "disabled"
+		case *connmgr.BasicConnMgr:
+			infos["type"] = "basic"
+
+			inf := cmgr.GetInfo()
+			infos["lowWater"] = inf.LowWater
```

Comment:
I assume these are supposed to be human readable keys? If so, we should probably just say things like "Low Watermark" instead.

Reply:
Those will be the result of JSON endpoint. We can create mappings in TextMarshaller.